src/docker-credentials.ts

```diff
@@ -0,0 +1,71 @@
+export class DockerCredential {
+  /**
+   * Create a credential for DockerHub. This method assumes that you will have long-lived
+   * Github Secrets stored under the usernameKey and personalAccessTokenKey.
+   */
+  public static dockerHub(creds: DockerHubCredentialSecrets = {}): DockerCredential {
+    return new DockerCredential(
+      'docker',
+      undefined,
+      creds.usernameKey ?? 'DOCKERHUB_USERNAME',
```

Comment:
Can you double-check these names are commonly used across the internet?

Reply:
This looks to be the correct defaults. It says so on the official [docker login action](https://github.com/docker/login-action#docker-hub), and comes up often when I search for them on google.